packages/shared-state-mesh_wide_upgrade/files/etc/shared-state/hooks/mesh_wide_upgrade/start_upgrade

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/lua
+
+--! LibreMesh
+--! Copyright (C) 2019  Gioacchino Mazzurco <gio@altermundi.net>
+--!
+--! This program is free software: you can redistribute it and/or modify
+--! it under the terms of the GNU Affero General Public License as
+--! published by the Free Software Foundation, either version 3 of the
+--! License, or (at your option) any later version.
+--!
+--! This program is distributed in the hope that it will be useful,
+--! but WITHOUT ANY WARRANTY; without even the implied warranty of
+--! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--! GNU Affero General Public License for more details.
+--!
+--! You should have received a copy of the GNU Affero General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+local JSON = require("luci.jsonc")
+local utils = require "lime.utils"
+local mesh_upgrade = require "lime.mesh-upgrade"
+
+
+--if already involved in an upgrade transaction do nothing
+if not mesh_upgrade.mesh_upgrade_is_started() then
+	for node, s_s_data in pairs(JSON.parse(io.stdin:read("*all"))) do
+		--if any node has started an upgrade process and start one too?
+		--only fetch the info from the master node publication?
+		if s_s_data.data.master_node == node then
+
+			mesh_upgrade.set_mesh_upgrade_info(s_s_data.data,mesh_upgrade.upgrade_states.STARTING)
+			utils.unsafe_shell('logger -p daemon.info -t "mesh upgrade" "mesh upgrade detected upgrade from"' ..
+			s_s_data.data.repo_url .. " " .. node)
+		end
+		utils.unsafe_shell('logger -p daemon.info -t "messwideupgrade" "hoock +++++ "' ..
```

Comment:
`hoock` maybe is a typo and should be hook?

Reply:
agree